dart/optimizer/Function.hpp

```diff
@@ -38,107 +38,108 @@
 
 #include <Eigen/Dense>
 
+#include "dart/common/Deprecated.hpp"
```

Comment:
It looks like we don't end up using `Deprecated.hpp`, presumably because there isn't really any good way of deprecating the non-const version of the function without a `const_cast`.